backend/app/graphql/onboarding_request.py

```diff
@@ -31,7 +28,71 @@ class OnboardingRequest(graphene.ObjectType):
     status = graphene.String()
 
 
+# Queries
+
+# Return object for queries
+class GetOnboardingRequest(Query):
+    contact_name = graphene.String()
+    contact_email = graphene.String()
+    contact_phone = graphene.String()
+    role = graphene.String()
+    date_submitted = graphene.DateTime()
+    status = graphene.String()
+
+
+class OnboardingRequestQueries(QueryList):
+    getAllOnboardingRequests = graphene.List(
+        GetOnboardingRequest,
+        first=graphene.Int(default_value=5),
+        offset=graphene.Int(default_value=0),
+        role=graphene.String(default_value=""),
+        status=graphene.String(default_value=""),
+    )
+    getOnboardingRequestById = graphene.List(
+        GetOnboardingRequest,
+        id=graphene.String(),
+        first=graphene.Int(default_value=5),
+        offset=graphene.Int(default_value=0),
+    )
+
+    def resolve_getAllOnboardingRequests(self, info, first, offset, role, status):
+        requests = services["onboarding_request_service"].get_all_onboarding_requests(
+            role, status
+        )
+        return [
+            *map(
+                lambda request: GetOnboardingRequest(
+                    contact_name=request.contact_name,
+                    contact_email=request.contact_email,
+                    contact_phone=request.contact_phone,
+                    role=request.role,
+                    date_submitted=request.date_submitted,
+                    status=request.status,
+                ),
+                requests[offset : offset + first],
```

Comment:
This looks good broadly, but unfortunately by putting the offset filter here and converting to dto, we're now iterating over ALL data before it's limited to first:first+offset. Would you be able to either move the parameters into the service as well (so we select [first:first+offset] before we iterate there) or else I guess just revert to the non-DTO version?